src/lighteval/utils/imports.py

```diff
@@ -11,158 +11,124 @@
 
 
-CANNOT_USE_EXTENDED_TASKS_MSG = "If you want to use extended_tasks, make sure you installed their dependencies using `pip install -e .[extended_tasks]`."
+def raise_if_package_not_available(package_name: str | Extras, *, language: str = None, object_name: str = None):
+    prefix = "You" if object_name is None else f"Through the use of {object_name}, you"
 
+    if package_name == Extras.MULTILINGUAL and (
+        (language is not None) or (not is_multilingual_package_available(language))
+    ):
+        raise ImportError(prefix + not_installed_error_message(package_name)[3:])
 
-def can_load_multilingual_tasks() -> bool:
-    try:
-        import lighteval.tasks.multilingual.tasks  # noqa: F401
+    if not is_package_available(package_name):
+        raise ImportError(prefix + not_installed_error_message(package_name)[3:])
 
-        return True
-    except ImportError:
-        return False
 
+def not_installed_error_message(package_name: str | Extras) -> str:
+    if package_name == Extras.MULTILINGUAL.value:
+        return "You are trying to run an evaluation requiring multilingual capabilities. Please install the required extra: `pip install lighteval[multilingual]`"
+    elif package_name == Extras.EXTENDED.value:
+        return "You are trying to run an evaluation requiring additional extensions. Please install the required extra: `pip install lighteval[extended] "
+    elif package_name == "text_generation":
+        return "You are trying to start a text generation inference endpoint, but TGI is not present in your local environment. Please install it using pip."
+    elif package_name in ["bitsandbytes", "auto-gptq"]:
+        return f"You are trying to load a model quantized with `{package_name}`, which is not available in your local environment. Please install it using pip."
+    elif package_name == "peft":
+        return "You are trying to use adapter weights models, for which you need `peft`, which is not available in your environment. Please install it using pip."
+    elif package_name == "openai":
+        return "You are trying to use an Open AI LLM as a judge, for which you need `openai`, which is not available in your environment. Please install it using pip."
 
-CANNOT_USE_MULTILINGUAL_TASKS_MSG = "If you want to use multilingual tasks, make sure you installed their dependencies using `pip install -e .[multilingual]`."
+    return f"You requested the use of `{package_name}` for this evaluation, but it is not available in your current environment. Please install it using pip."
 
 
-def can_load_spacy_tokenizer(language: str) -> bool:
-    imports = []
-    packages = ["spacy", "stanza"]
-    if language == "vi":
-        packages.append("pyvi")
-    elif language == "zh":
-        packages.append("jieba")
-
-    for package in packages:
-        imports.append(importlib.util.find_spec(package))
-    return all(cur_import is not None for cur_import in imports)
-
+class DummyObject(type):
```

Comment:
Can you explain what it's for? Is it only for tests?

Reply:
It's a metaclass for all "dummy objects". Dummy objects are placeholders that will raise an import error as soon as you try to access any method tied to it.

This basically allows defining any class with any dependency; as you assign a `@requires` decorator to it, it will check the required dependencies. If you don't have the required dependencies installed, the decorator will return such placeholders instead of the original class.

The current doc is the following:
```
    Metaclass for the dummy objects. Any class inheriting from it will return the ImportError generated by
    `requires_backend` each time a user tries to access any method of that class.
```

Let me know if unclear and happy to provide some more

Reply:
It's a bit clearer but 'm still unsure in which case I should use this